### PR TITLE
Simplify cross compilation with CUDA for ARM boards. 

### PR DESCRIPTION
**Pull request by andy-held**
* CUDA arch is set for all ARM targets
* --unresolved-symbols=ignore-in-shared-libs is set when cross-compiling for ARM
* FindCUDA.cmake was adjusted to find shared library stubs within the CUDA toolkit.

### CODE REVIEW

**Comment by paroj** [comment]
:+1:

**Comment by ilya-lavrenov** [comment]
Which problem this change solves?

**Comment by andy-held** [comment]
1. CMake does not correctly find the necessary CUDA libraries when cross compiling for the NVIDIA Jetson TK1. The NVIDIA CUDA cross compilation toolkit for Linux4Tegra comes with .so stubs. They are located in a different directory and to use them the '--unresolved-symbols=ignore-in-shared-libs' compiler option has to be specified.
3. CUDA_ARCH_BIN was only set for ARM devices when ANDROID was defined in CMake.

File: cmake/FindCUDA.cmake [hunk 1]
```diff
@@ -685,6 +685,9 @@ macro(cuda_find_library_local_first_with_path_ext _var _names _doc _path_ext )
     # and old paths.
     set(_cuda_64bit_lib_dir "${_path_ext}lib/x64" "${_path_ext}lib64" "${_path_ext}libx64" )
   endif()
+  if(CMAKE_CROSSCOMPILING AND ARM)
```
Comment (ilya-lavrenov) [inline cmake/FindCUDA.cmake]: could you please fix the same for `AARCH64`?

**Comment by andy-held** [comment]
CUDA cross compilation toolkit library folder is now also set correctly on AARCH64.

File: cmake/OpenCVDetectCUDA.cmake [hunk 2]
```diff
@@ -200,6 +198,10 @@ if(CUDA_FOUND)
       set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} -Xcompiler -fno-finite-math-only)
     endif()
 
+    if(CMAKE_CROSSCOMPILING AND ARM)
```
Comment (ilya-lavrenov) [inline cmake/OpenCVDetectCUDA.cmake]: and here.

**Comment by andy-held** [comment]
Fixed it.
Very thorough review and good feedback. Thanks.

**Comment by jet47** [comment]
:+1:

**Pull request merged**